### PR TITLE
perf: prevent setting all languages for all apps on initial load

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Changes the way how app languages get set in `gettext` so only the current language is being considered initially. Other languages will be set dynamically when changing the current language.

Also refactors the language setup process a bit to make it simpler.

`announceGettext` sets up gettext in our Vue app, without any translations yet though. `announceTranslations` gets called later during the bootstrap process and provides all translations to `gettext`: the ones for core, additional ones and the ones for the apps for the current language. That way we have all that stuff in one place.

## Related Issue

- refs https://github.com/owncloud/web/issues/11035

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: changelog/unreleased/enhancement-page-loading-times [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Enhancement: Faster page loading times
+
+We've massively increased the initial page loading times. This is especially noticiable with slower network connections.
```
Comment (phil-davis) [inline changelog/unreleased/enhancement-page-loading-times:3]:
```suggestion
We've massively decreased the initial page loading times. This is especially noticeable with slower network connections.
```

File: packages/web-runtime/src/helpers/language.ts [hunk 2]
```diff
@@ -16,3 +18,31 @@ export const setCurrentLanguage = ({
     document.documentElement.lang = currentLanguage
   }
 }
+
+/**
+ * Loads all app translations for one given language.
+ * This is adviced to call each time the language is being changed.
```
Comment (phil-davis) [inline packages/web-runtime/src/helpers/language.ts:24]:
```suggestion
 * This should be called each time the language is being changed.
```

File: changelog/unreleased/enhancement-page-loading-times (hunk 1 above)
Comment (JammingBen) [reply changelog/unreleased/enhancement-page-loading-times:3]: Haha 😂 Thanks for the hint!

File: changelog/unreleased/enhancement-page-loading-times [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+According to the lighthouse performance metrics, we improved from a score of 60 to 94.
+
+This has been achieved by taking several different actions:
+
+- Lazy load large dependencies
+- Improve chunking
+- Enable server side compression
+- Reduce the size of the main font we're using
+- Stop setting all app languages on the initial page load
```
Comment (dschmidt) [inline changelog/unreleased/enhancement-page-loading-times:13]: Can we improve this wording? As a user I have no idea what this means

File: packages/web-runtime/src/container/bootstrap.ts [hunk 4]
```diff
@@ -382,13 +364,26 @@ export const announceTranslations = ({
...
+  customTranslations?: Translations
+  appsStore?: AppsStore
+}) => {
+  gettext.translations = merge(coreTranslations, customTranslations || {})
+
+  if (appsStore) {
+    loadAppTranslations({
+      apps: appsStore.apps,
+      gettext,
+      lang: gettext.current
```
Comment (dschmidt) [inline packages/web-runtime/src/container/bootstrap.ts:384]: Is this correctly set on application boot? When is the language setting loaded?

File: packages/web-runtime/src/helpers/language.ts [hunk 5]
```diff
@@ -16,3 +18,31 @@ export const setCurrentLanguage = ({
...
+}) => {
+  const appTranslations = Object.values(apps).reduce<Translations>((acc, app) => {
+    const { translations } = app
+
+    if (gettext.translations[lang] && translations[lang]) {
+      acc = merge(acc, translations[lang])
+    }
+
+    return acc
+  }, {})
```
Comment (dschmidt) [inline packages/web-runtime/src/helpers/language.ts:43]: It's my personal taste, but I always feel it's weird to use `reduce` in this way. This is basically the same as:

```
let acc: Translations = {}
Object.values(apps).forEach(app => {
    const { translations } = app
    if (gettext.translations[lang] && translations[lang]) {
      acc = merge(acc, translations[lang])
    }
})
```
isn't it?

**Review by dschmidt — COMMENTED** [review]
LGTM 😍 

Haven't tested it, but code wise looks so much better than before

File: packages/web-runtime/src/helpers/language.ts (hunk 5 above)
Comment (dschmidt) [reply packages/web-runtime/src/helpers/language.ts:43]: possibly could also be `const acc` and then just `Object.assign` instead of merge

Comment (JammingBen) [reply packages/web-runtime/src/helpers/language.ts:43]: Yup that works as well. No hard feelings from my side 🤷

File: packages/web-runtime/src/container/bootstrap.ts (hunk 4 above)
Comment (JammingBen) [reply packages/web-runtime/src/container/bootstrap.ts:384]: We need to set it here initially to make sure the correct language has been loaded in an unauthenticated context. In an authenticated context, we set it a second time after the user has been fetched via the Graph API.

File: packages/web-runtime/src/helpers/language.ts [hunk 6]
```diff
@@ -16,3 +18,29 @@ export const setCurrentLanguage = ({
...
+  lang
+}: {
+  apps: Record<string, ApplicationInformation>
+  gettext: Language
+  lang: string
+}) => {
+  const appTranslations: Translations = {}
+  Object.values(apps).forEach((app) => {
+    const { translations } = app
+    if (gettext.translations[lang] && translations[lang]) {
```
Comment (kulmann) [inline packages/web-runtime/src/helpers/language.ts:38]: What is the `gettext.translations[lang]` check for here? 🙈

Comment (JammingBen) [reply packages/web-runtime/src/helpers/language.ts:38]: As discussed: It checks if we support the requested language and that there are translations available for it. This is basically determined by the core translations.

**Review by dschmidt — APPROVED** [review]

**Pull request merged**